tests/unit/integrations/watchers/indexable-attachment-watcher-test.php

```diff
@@ -0,0 +1,252 @@
+<?php
+
+namespace Yoast\WP\SEO\Tests\Unit\Integrations\Watchers;
+
+use Brain\Monkey;
+use Mockery;
+use Yoast_Notification_Center;
+use Yoast\WP\SEO\Tests\Unit\TestCase;
+use Yoast\WP\SEO\Config\Indexing_Reasons;
+use Yoast\WP\SEO\Helpers\Indexing_Helper;
+use Yoast\WP\SEO\Integrations\Cleanup_Integration;
+use Yoast\WP\SEO\Helpers\Attachment_Cleanup_Helper;
+use Yoast\WP\SEO\Conditionals\Migrations_Conditional;
+use Yoast\WP\SEO\Actions\Indexing\Indexable_Post_Indexation_Action;
+use Yoast\WP\SEO\Integrations\Watchers\Indexable_Attachment_Watcher;
+
+
+/**
+ * Class Indexable_Attachment_Watcher_Test.
+ *
+ * @group indexables
+ * @group integrations
+ * @group watchers
+ *
+ * @coversDefaultClass Yoast\WP\SEO\Integrations\Watchers\Indexable_Attachment_Watcher
+ * @covers Yoast\WP\SEO\Integrations\Watchers\Indexable_Attachment_Watcher
+ */
+class Indexable_Attachment_Watcher_Test extends TestCase {
+
+	/**
+	 * The indexing helper.
+	 *
+	 * @var Indexing_Helper
+	 */
+	protected $indexing_helper;
+
+	/**
+	 * The attachment cleanup helper.
+	 *
+	 * @var Attachment_Cleanup_Helper
+	 */
+	protected $attachment_cleanup;
+
+	/**
+	 * The notifications center.
+	 *
+	 * @var Yoast_Notification_Center
+	 */
+	protected $notification_center;
+
+	/**
+	 * The Indexable_Attachment_Watcher instance.
+	 *
+	 * @var Indexable_Attachment_Watcher
+	 */
+	protected $instance;
+
+	/**
+	 * Sets up the tests.
+	 */
+	protected function set_up() {
+		parent::set_up();
+
+		$this->indexing_helper     = Mockery::mock( Indexing_Helper::class );
+		$this->attachment_cleanup  = Mockery::mock( Attachment_Cleanup_Helper::class );
+		$this->notification_center = Mockery::mock( Yoast_Notification_Center::class );
+
+		$this->instance = new Indexable_Attachment_Watcher(
+			$this->indexing_helper,
+			$this->attachment_cleanup,
+			$this->notification_center
+		);
+	}
+
+	/**
+	 * Tests the get_conditionals method.
+	 *
+	 * @covers ::get_conditionals
+	 */
+	public function test_get_conditionals() {
+		$this->assertSame( [ Migrations_Conditional::class ], $this->instance->get_conditionals() );
+	}
+
+	/**
+	 * Tests the register_hooks method.
+	 *
+	 * @covers ::register_hooks
+	 */
+	public function test_register_hooks() {
+
+		Monkey\Actions\expectAdded( 'update_option_wpseo_titles' )
```

Comment:
I have also seen $this->assertNotFalse(has_action(.... not sure if there is a preferred method, otherwise I would suggest that we keep using the assertNotFalse since we use it all over te place?

Reply:
Done!